### PR TITLE
[ASENG-814] Replace GDOTV as preferred UI

### PR DESCRIPTION
**Pull request by d0g0x01**
Replace GDOTV as preferred UI with AWS graph notebook given recent licensing changes

### CODE REVIEW

**Review by Minosity-VR — APPROVED** [review]
Nice, thanks a lot for finding an alternative to gdotv!
I really like the notebooks, quite useful to keep an history of previous queries by creating new cells, and it can probably be paired with some python scripting to manipulate the data once we extracted what we want (but that's for another step 😁)

Just a question about the color scheme, do you think it could be possible to create a mapping with expose/severity? I feel like here the mapping is done through the number of steps, I remember seeing some yellow critical node when doing the tests with you last week, it could be nice to always have a consistent coloring :D

File: deployments/kubehound/notebook/shared/shared.ipynb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    "  \"ssl\": false,\n",
+    "  \"gremlin\": {\n",
+    "    \"traversal_source\": \"g\",\n",
+    "    \"username\": \"\",\n",
+    "    \"password\": \"\",\n",
+    "    \"message_serializer\": \"graphsonv3\"\n",
+    "  }\n",
+    "}"
+   ]
+  },
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/shared/shared.ipynb:38]: done

File: deployments/kubehound/notebook/SecurityPosture.ipynb [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    "kh.V().\n",
+    "    hasLabel(\"Container\").\n",
+    "    count().\n",
+    "    aggregate(\"t\").\n",
+    "    V().\n",
+    "    hasLabel(\"Container\").\n",
+    "    hasCriticalPath().\n",
+    "    count().\n",
+    "    as(\"e\").\n",
+    "    math(\"e/t\").by().by(unfold())"
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/SecurityPosture.ipynb:167]: done

File: deployments/kubehound/notebook/SecurityPosture.ipynb [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    "    has(\"critical\", false).\n",
+    "    count().\n",
+    "    aggregate(\"t\").\n",
+    "    V().\n",
+    "    hasLabel(\"Identity\").\n",
+    "    has(\"critical\", false).\n",
+    "    hasCriticalPath().\n",
+    "    count().\n",
+    "    as(\"e\").\n",
+    "    math(\"e/t\").by().by(unfold())"
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/SecurityPosture.ipynb:141]: done

**Comment by d0g0x01** [comment]
> Nice, thanks a lot for finding an alternative to gdotv! I really like the notebooks, quite useful to keep an history of previous queries by creating new cells, and it can probably be paired with some python scripting to manipulate the data once we extracted what we want (but that's for another step 😁)
> 
> Just a question about the color scheme, do you think it could be possible to create a mapping with expose/severity? I feel like here the mapping is done through the number of steps, I remember seeing some yellow critical node when doing the tests with you last week, it could be nice to always have a consistent coloring :D

it depends on the cell but most mappings are done by severity. It's just in some cases you expect no critical assets so I've done mapping by type. Is there an example you found it missing?

File: deployments/kubehound/notebook/BlueTeam.ipynb [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    "    by(elementMap())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Blast radius evaluation\n",
+    "\n",
+    "It may be the case that the compromised container does not have a path to a critical asset (at least within the KubeHound model). In this case in can be useful to understand the blast radius of the compromised container. We can use KubeHound to identify all the resources that an attacker could have accessed from a compromised container."
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/BlueTeam.ipynb:282]: It generates a lot of data though. Using critical paths should be the first step as this automatically filters less interesting paths to focus the investigation

File: deployments/kubehound/docker-compose.dev.yaml [hunk 5]
```diff
@@ -15,20 +15,6 @@ services:
     volumes:
       - kubegraph_data:/var/lib/janusgraph
 
-  mongo-express:
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/docker-compose.dev.yaml:18]: removed

File: deployments/kubehound/notebook/BlueTeam.ipynb [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    "\tdedup()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Shortest attack paths\n",
+    "\n",
+    "Attackers are incentivized to take the shortest path to their target. We can use KubeHound to identify the shortest attack paths from the compromised container to the critical assets. This can help us focus our detection efforts on the most likely attack paths.\n",
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/BlueTeam.ipynb:224]: distribution would be crazy expensive query so probably want to leave that out. The shortest path is just used as a guide, the user can just change the number in the next cell

File: deployments/kubehound/notebook/SecurityPosture.ipynb [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%gremlin\n",
+    "kh.V().\n",
+    "    hasLabel(\"Endpoint\").\n",
+    "    has(\"exposure\", gte(2)).\n",
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/SecurityPosture.ipynb:104]: added a link to the doc

**Comment by Minosity-VR** [comment]
> it depends on the cell but most mappings are done by severity. It's just in some cases you expect no critical assets so I've done mapping by type. Is there an example you found it missing?

I just recalled the discussion we had last week when writing the overall comment, but I may misremember something. If mapping is done by severity then perfect!

File: deployments/kubehound/notebook/BlueTeam.ipynb [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+    "        has(\"type\", \"User\").has(\"name\", \"bits.barkley@datadoghq.com\")).\n",
+    "    hasCriticalPath().\n",
+    "    values(\"name\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's examine the possible attack paths that could be taken by an attacker who has access to the compromised credentials."
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/BlueTeam.ipynb:108]: done (with the OR)

File: deployments/kubehound/notebook/BlueTeam.ipynb [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compromised Credentials\n",
+    "\n",
+    "Let us consider a scenario where a user's credentials have been compromised. We can use KubeHound to identify the resources that the user has access to, whether any lead to critical assets and what attacks might have been leveraged.\n",
+    "\n",
+    "First let's see whether there are any critical paths accessible. Because Kubernetes delegates the management of users' group memberships to third party components (e.g identity providers), we need to check paths from both the user and any groups they are a member of."
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/BlueTeam.ipynb:86]: added a comment

File: deployments/kubehound/notebook/KubeHound.ipynb [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    "    by(elementMap()).\n",
+    "    limit(100)  // Limit the number of results for large clusters"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Complex critical path queries\n",
+    "The KubeHound DSL is fully compatible with the Gremlin language, so you can use any gremlin function in your queries to add additional filters if needed. The example below attempts to filter for Kubernetes services exposed on port 443 attached to `elasticsearch` containers, and limits attack paths to 6 hops or less."
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/KubeHound.ipynb:131]: done

File: deployments/kubehound/notebook/KubeHound.ipynb [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### All critical paths from external services\n",
+    "\n",
+    "Critical paths from external services can highlight most likely paths an attacker can take following the compromise of an external service. In a well-configured cluster, such paths should be limited to a very few (if any) services.\n"
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/KubeHound.ipynb:88]: added a comment

File: deployments/kubehound/notebook/RedTeam.ipynb [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+    "## Worfklow"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initial Recon\n",
+    "\n",
+    "Look for exposed endpoints attached to containers and return the port details and image. This enables us to match against any exploits in our catalogue"
```
[the remark this answers is not in the document]
Comment (d0g0x01) [reply deployments/kubehound/notebook/RedTeam.ipynb:84]: this is just an example of a grouping I find useful. the notebook allows search and ordering itself. I guess you are wanting a different grouping but I think what is there at the moment is the most useful showcase

**Review by Minosity-VR — APPROVED** [review]

**Pull request merged**